pkg/adapter/apiserver/delegate.go

```diff
@@ -18,51 +18,69 @@ package apiserver
 
 import (
 	"context"
+	"sync"
 
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 	"github.com/google/uuid"
 	"go.uber.org/zap"
 	"k8s.io/client-go/tools/cache"
 	"knative.dev/eventing/pkg/adapter/apiserver/events"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	brokerfilter "knative.dev/eventing/pkg/broker/filter"
+	"knative.dev/eventing/pkg/eventfilter"
+	"knative.dev/eventing/pkg/eventfilter/subscriptionsapi"
 )
 
 type resourceDelegate struct {
 	ce                  cloudevents.Client
 	source              string
 	ref                 bool
 	apiServerSourceName string
+	filters             []eventingv1.SubscriptionsAPIFilter
 
 	logger *zap.SugaredLogger
 }
 
 var _ cache.Store = (*resourceDelegate)(nil)
+var filterSync sync.Once
+var filter eventfilter.Filter
```

Comment:
I think it would be cleaner by removing the globals and having a filter field in `resourceDelegate` of type `filter.Filter` with the pre-materialized `Filter` that assigned when creating the `resourceDelegate` :

```golang
rd := resourceDelegate{
    filter: subscriptionsapi.NewAllFilter(brokerfilter.MaterializeFiltersList(a.logger.Desugar(), filters)...),
    // ...
}
```
and then in `handleKubernetesObject` we would use/call the filter `a.filter(...)`

Reply:
I went for the easy route there and now I'm paying: I can't test it properly. I will give your approach a try, thanks for the suggestion.